tools/exec_trace_parser.py

```diff
@@ -64,48 +136,67 @@ def get_sfr_name(self, value):
             return "SFR @ 0x%08X" % sfr_addr
 
     def trace_version(self, value):
+        """Translate a TRACE_ExecTracerVersion() trace to human readable output."""
         ver_char = (value >> 16) & 0xFF
         ver_major = (value >> 8) & 0xFF
         ver_minor = (value >> 0) & 0xFF
         self.reset_indent()
         print("Tracer version %c%d.%d" % (ver_char, ver_major, ver_minor))
 
     def trace_reset(self, value):
+        """Translate a TRACE_ProcessorReset() trace to human readable output."""
         self.print_indent()
         print("Processor reset: 0x%02X" % value)
 
     def trace_func_entry(self, value):
+        """Translate a TRACE_FunctionEntry() trace to human readable output."""
         self.print_indent()
         print("Enter %s" % self.get_func_name(value))
         # Increment indent after Enter statement
         # This is like an opening brace
         self.inc_indent()
 
     def trace_func_exit(self, value):
+        """Translate a TRACE_FunctionExit() trace to human readable output."""
         # Decrement indent before Exit statement
         # This is like a closing brace
         self.dec_indent()
         self.print_indent()
         print("Exit %s" % self.get_func_name(value))
 
     def trace_file_and_line(self, value):
+        """Translate a TRACE_Line() trace to human readable output."""
         module_num = (value >> 16) & 0xFFF
         line_num = (value >> 0) & 0xFFFF
         self.print_indent()
         print("Module: %u, Line: %u" % (module_num, line_num))
 
     def trace_variable(self, addr_value, var_value):
+        """Translate a TRACE_VariableValue() trace to human readable output."""
         self.print_indent()
         print("%s = %d" % (self.get_var_name(addr_value), var_value))
 
     def trace_sfr(self, addr_value, reg_value):
+        """Translate a TRACE_SFRValue() trace to human readable output."""
         self.print_indent()
         print("%s = 0x%08X" % (self.get_sfr_name(addr_value), reg_value))
 
     def read_and_trace_next(self, trace_reader: TraceReaderInterface):
+        """Read the next value from the trace reader and translate it to human
```

Comment:
Are you using 'trace' to indicate that something is human readable? If yes, ensure that is in the README.

Reply:
I'm having trouble here with how to best use terminology.  The `trace_reader` object provides raw trace buffer values.  This functions retrieves the next raw trace buffer value, parses it and spits out human readable text to stdout.

The README is not done and will have a more complete theory of operation including diagrams.